base/services/umpnpmgr/rpcserver.c

```diff
@@ -2588,8 +2591,11 @@ PNP_GetInterfaceDeviceListSize(
     DPRINT("PNP_GetInterfaceDeviceListSize(%p %p %p %S 0x%08lx)\n",
            hBinding, pulLen, InterfaceGuid, pszDeviceID, ulFlags);
 
-    if (!IsValidDeviceInstanceID(pszDeviceID))
-        return CR_INVALID_DEVINST;
+    if (pszDeviceID != NULL && pszDeviceID[0] != L'\0')
```

Comment:
```suggestion
    if (pszDeviceID && *pszDeviceID)
```

Reply:
So, it's valid for PNP_GetInterfaceDeviceListSize to get a NULL pszDeviceID ?

Reply:
from CM_Get_Device_Interface_List docs:
"If this value(pszDeviceID) is NULL, or if it points to a zero-length string, the function retrieves all interfaces that belong to the specified class."
CM_Get_Device_Interface_List calls CM_Get_Device_Interface_List_ExW which calls PNP_GetInterfaceDeviceList
From other side, PNP_GetInterfaceDeviceList calls NtPlugPlayControl(PlugPlayControlGetInterfaceDeviceList) which actually IoGetDeviceInterfaces and pszDeviceID is used to got a DeviceObject for this function, but Device Object can be NULL, ass well as pszDeviceID

Reply:
From docs
The CM_Get_Device_Interface_List_Size function retrieves the buffer size that must be passed to the CM_Get_Device_Interface_List